src/RecursiveArrayTools.jl

```diff
@@ -126,6 +126,9 @@ additional properties:
 An AbstractDiffEqArray adds the following fields:
 
 * `t` which holds the times of each timestep.
+* `discretes` (optional) which contains a
+  `SymbolicIndexingInterface.ParameterTimeseriesCollection` to allow storing and indexing
+  discrete variables. Indexing behavior is handled via `SymbolicIndexingInterface.getu`.
```

Comment:
There should be a mention of symbolic indexing for DiffEqArray in a `# Symbolic Array Interface` section? And any other requirements you implicitly have put on DiffEqArray to make that work should get mentioned.